HLTrigger/HLTcore/interface/HLTConfigProvider.h

```diff
@@ -186,11 +187,9 @@ class HLTConfigProvider {
   /// HLT prescale value in specific prescale set for a specific trigger path
   template <typename T = unsigned int>
   T prescaleValue(unsigned int set, const std::string& trigger) const {
-    //limit to only 4 allowed types
-    static_assert(std::is_same_v<T, unsigned int> or std::is_same_v<T, FractionalPrescale> or std::is_same_v<T, int> or
-                      std::is_same_v<T, double>,
-                  "Please use prescaleValue<unsigned int>, prescaleValue<int>, prescaleValue<double>, or "
-                  "prescaleValue<FractionalPrescale>,\n note int and unsigned int will be depreated soon");
+    static_assert(std::is_same_v<T, double> or std::is_same_v<T, FractionalPrescale>,
+                  "\n\tPlease use prescaleValue<double> or prescaleValue<FractionalPrescale>"
+                  "\n\t(other types for HLT prescales are not supported anymore by HLTConfigProvider");
     return hltConfigData_->prescaleValue(set, trigger);
   }
 
```

Comment:
https://github.com/cms-sw/cmssw/blob/d8411f0d619b76ba4ccac946e51fb0ec9f3aef55/HLTrigger/HLTcore/interface/HLTConfigProvider.h#L198

With this call, a user might still access HLT prescales as `uint` via `HLTConfigProvider`, with `HLTConfigProvider::prescaleTable().prescaleValue(..)`. This goes against the goal of this PR, so I'm trying to find a solution, possibly without updating the `HLTPrescaleTable` class in `DataFormats/HLTReco`.

Reply:
Is this accessor used anywhere? If not we can just remove it.

Reply:
OK, circular argument. So it seems like HLTConfigData needs to be updated as well.

Reply:
> So it seems like HLTConfigData needs to be updated as well.

Done in the last push.

> Is this accessor used anywhere? If not we can just remove it.

I think it is used only by `PATTriggerProducer`, and even there it is not really necessary. Still, I think it makes sense to make it accessible via `HLTConfigData`. To do that with `double` and `FractionalPrescale`, I found no better way than adding two copies of the PS table (one per type) in `HLTConfigData`, and specialising the `prescaleTable<T>` method.

I have extended the unit test in `test_catch2_HLTConfigData.cc` with some simple checks on `prescaleTable<double>` and `prescaleTable<FractionalPrescale>`.